### PR TITLE
Table v2, Redux Refactor

### PR DESCRIPTION
**Pull request by kevinli-work**
* Includes a brand new rewritten table library with all new API (includes docs in `components/sharedComponents/IBTable/README.md`)
* Removed 10 top-level Redux stores and many many child elements
* Removed many supporting Redux actions
* Fully refactored many tests to use the new `async/await` and `jest.fn()` class function mocking patterns

https://federal-spending-transparency.atlassian.net/browse/DS-1993

### CODE REVIEW

File: src/js/components/search/table/cells/ResultsTableHeaderCell.jsx [hunk 1]
```diff
@@ -1,112 +1,104 @@
...
-    clickedHeader() {
-        // check if this is the field that is currently being used to sort
-        if (this.props.column === this.props.order.field) {
-            // it's the same field, just toggle the direction
-            let direction = 'asc';
-            if (this.props.order.direction === 'asc') {
-                direction = 'desc';
-            }
+    const clickedDefault = () => {
+        // if (props.isActive) {
```
Comment (mab6bf) [inline src/js/components/search/table/cells/ResultsTableHeaderCell.jsx]: Do we need this commented-out code?

File: src/js/components/sharedComponents/IBTable/IBTable.jsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /**
   * IBTable.jsx
-  * Created by Kevin Li 12/12/16
+  * Created by Kevin Li 12/8/16
```
Comment (mab6bf) [inline src/js/components/sharedComponents/IBTable/IBTable.jsx]: 12/8/17?

File: src/js/components/sharedComponents/IBTable/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+# IBTable
```
Comment (mab6bf) [inline src/js/components/sharedComponents/IBTable/README.md]: +1 for documentation

**Comment by mab6bf** [comment]
Also, that tabbing issue we discussed

File: src/js/components/search/table/cells/ResultsTableHeaderCell.jsx (hunk 1 above)
Comment (kevinli-work) [reply src/js/components/search/table/cells/ResultsTableHeaderCell.jsx]: Yes, this is the old code that enables ascending sort, for when the backend is capable of doing it again.

**Pull request merged**